### PR TITLE
[MIG] sale_commission: Migrated to 10.0

### PR DESCRIPTION
**Pull request by hurrinico**
Porting of module sale_commission to Version 10

### CODE REVIEW

**Comment by hurrinico** [comment]
Still WIP

File: sale_commission/models/sale_commission.py
Comment (tafaRU) [inline sale_commission/models/sale_commission.py:51]: commission I think it's more suitable than section, what do you think?

File: sale_commission/models/sale_commission.py
Comment (tafaRU) [inline sale_commission/models/sale_commission.py]: s/self/section(or commission)

**Review by tafaRU — CHANGES_REQUESTED** [review]
In addition please note the `<data>` tag in xml files can be removed.

Thanks!

File: sale_commission/models/sale_commission.py
Comment (hurrinico) [reply sale_commission/models/sale_commission.py:51]: this model isn't sale commission but sale.commission.section, so maybe section is correct.

**Comment by pedrobaeza** [comment]
Hi, I'm going to work these days on finishing 9.0 migration with the latest changes including the web_widget_one2many_tags improvement, so maybe it's better to wait a bit for this migration and review the others. What do you think?

**Comment by hurrinico** [comment]
@pedrobaeza i agree. When you finish the web_widget_one2many_tags porting we can improve this one and merge it

**Comment by eLBati** [comment]
As workaround for https://github.com/odoo/odoo/issues/17618 I propose https://github.com/hurrinico/commission/pull/2

**Review by rgarnau — APPROVED** [review]
We have made an improvement for this PR in order to prepare agents' data outside from the method recompute_lines_agents.
See https://github.com/hurrinico/commission/pull/4.

@jbeficent

**Review by eLBati — CHANGES_REQUESTED** [review]
@hurrinico please fix
```
sale_commission/models/account_invoice.py:55:1: W293 blank line contains whitespace
```
and squash merge commits

**Comment by hurrinico** [comment]
@eLBati travis is green

**Review by JordiBForgeFlow — APPROVED** [review]

File: sale_commission/models/sale_commission.py
Comment (tafaRU) [reply sale_commission/models/sale_commission.py:51]: ok, that's right!

**Comment by pedrobaeza** [comment]
Hi, guys, sorry for not answering about the update on 9.0, but I haven't been able to do it. This is still valid if there's no bug embedding a one2many inside a one2many (on v8 it was).

**Comment by hurrinico** [comment]
@pedrobaeza there is a bug opened by @eLBati  odoo/odoo#17618, that is fixed by https://github.com/hurrinico/commission/pull/2

**Comment by eLBati** [comment]
Hi @rgarnau , did you configure your github account?
I see in commit
https://github.com/OCA/commission/pull/117/commits/0ddb38650042800a67216483a9179703fbdec45c
you are not recognized.
Thanks

**Comment by rgarnau** [comment]
Hi @eLBati - yes, already fixed. However, since this commit is already merged, I cannot amend the author anymore...

**Comment by pedrobaeza** [comment]
Thanks for the info. It's still a bit uglier than with web_widget_one2many_tags, but it if works, perfect to continue this way.

**Review by tafaRU — APPROVED** [review]

**Comment by hurrinico** [comment]
@rgarnau fixed.
i had conflicts try to fix flake 8 on your pr, so i made that mistake

**Comment by eLBati** [comment]
@rgarnau about your commits in this PR, github is saying
> Unrecognized author. If this is you, make sure rgarnau@gmail.com is associated with your account. Click to add email addresses in your account settings

**Comment by rgarnau** [comment]
@eLBati As I don't have permissions in this repo, I cannot amend the author of my commits. It's already fixed in my pull request.

**Comment by eLBati** [comment]
@rgarnau is rgarnau@gmail.com  associated with your account?

**Comment by rgarnau** [comment]
@eLBati no, that was a mistake. Instead, my account is rgarnau20@gmail.com

**Comment by eLBati** [comment]
Ok, so @hurrinico please edit commits form @rgarnau fixing the email address. Thanks

**Comment by eLBati** [comment]
@hurrinico @rgarnau done

**Comment by eLBati** [comment]
Ok, fixed quaterly settlement with https://github.com/OCA/commission/pull/117/commits/9279e38a7038a06f423b23f6ba5a66f98af660bc

**Review by eLBati — APPROVED** [review]

**Review by atchuthan — APPROVED** [review]
Functional Test in Runbot :+1:

**Comment by hurrinico** [comment]
@pedrobaeza @eLBati @atchuthan i can't remember what remains to do here

**Comment by pedrobaeza** [comment]
@hurrinico nothing more as I can see. Just squash your last 3 commits together and then I'll merge.

**Comment by eLBati** [comment]
@pedrobaeza I squashed the last 2 commits only, to preserve authorship

**Comment by eLBati** [comment]
@pedrobaeza please don't merge because we found other issues with ```agents``` o2m

**Comment by hurrinico** [comment]
IMHO after @eLBati commit this pr can be merged

**Comment by atchuthan** [comment]
I agree with @hurrinico suggestion to merge this PR.
We could add the fix for o2m in a new PR

**Review by JordiBForgeFlow — CHANGES_REQUESTED** [review]
We have issues when creating a demo database when the demo data tries to load a record expecting languange en_US to exist. See:
https://github.com/OCA/commission/pull/117/files#diff-855c05c79944f0f9cab407e8e2451171R51

**Comment by eLBati** [comment]
@jbeficent do you suggest to remove line(s)
```
<field name="lang">en_US</field>
```
?

**Comment by atchuthan** [comment]
@eLBati 
you could remove the line

Case scenario to reproduce:
- create demo database with another language except for en_US
- install this module and it will provide error

**Comment by hurrinico** [comment]
@atchuthan i've removed the lines

**Review by atchuthan — APPROVED** [review]
:+1:

@jbeficent: please check and review your test case

**Comment by pedrobaeza** [comment]
@jbeficent's case is covered already, so I merge this one finally.

**Pull request merged**
